lib/factory.rb

```diff
@@ -1 +1,94 @@
 # frozen_string_literal: true
+class Factory
+  def self.new(*attributes, &block)
+    class_name_from_string = attributes.shift.capitalize if attributes.first.is_a?(String)
+    class_name = Class.new do
+      attributes.each do |attribute|
+        attr_accessor attribute
+      end
+
+      define_method :initialize do |*value|
+        raise ArgumentError if value.length > attributes.length
+        attributes.each_with_index do |attribute, index|
+          instance_variable_set("@#{attribute}", value[index])
+        end
+      end
+
+      def [](attribute)
+        if attribute.is_a? Numeric
+          raise IndexError unless instance_variables[attribute.floor]
+          instance_variable_get("@#{members[attribute]}")
+        elsif attribute.is_a? Float
+          instance_variable_get("@#{members[attribute.to_i]}")
+        else
+          raise NameError unless members.include?(attribute.to_sym)
+          instance_variable_get("@#{attribute}")
+        end
+      end
+
+      def []=(attribute, value)
+        if attribute.is_a? Integer
+          raise IndexError unless instance_variables[attribute]
+        end
+        raise NameError unless instance_variable_get("@#{attribute}")
```

Comment:
What if `instance_variable_get` returns nil?

Reply:
If instance_variable_get returns nil error does not raise.